### ISSUE TITLE
Validate field with the same names

### ISSUE THREAD

**Issue by dmitriytat** [dmitriytat/jedi-validate#3]
Now it can not validate inputs with same names. 

Let me show an example of this issue.

We have a two phone inputs, with the same name.
`<input name="phone[]" required>`  - main phone
`<input name="phone[]" >` - additional phone

Now we can get around with this names 
`<input name="phone[0]" required>`  - main phone
`<input name="phone[1]" >` - additional phone

But I think it is not elegant.

**Comment by madcampos** [comment]
Hey, can i help with this?

**Comment by dmitriytat** [comment]
@madcampos yep, you can) I have only one idea (with out breaking changes). 
We need to add index to name automatically.

**Comment by madcampos** [comment]
I think this can be done like express' default parameters, like, if you repeat a field in a querystring it will create an array with the values instead of a simple value.
e.g.:

``` javascript
//the quesystring has ?phone=1&phone=2
//will output something like this:
{
    "phone": [1, 2]
}
```

**Comment by dmitriytat** [comment]
@madcampos We can't use the same input names in html without brackets []. Now we can send array like in your example (with brackets).

Issue about defining rules for elements with brackets. Rules and other staff are wrote in objects which keys are names on inputs.

```
{
  "phone[0]": {...}
  "phone[1]": {...}
  "email[]": {...}
  "email[]": {...}
}
```

and different emails can use different rules such as required for additional.

**Comment by madcampos** [comment]
I see, the diferent validations is indeed an issue for an array and most of the solutions of that aproach are not much elegants.
As of the `name` attribute, I think it behaves like with radio and checkboxes, or at least we can use that kind of logic.

One possible solution is to add automatic numbering, like this:

``` javascript
{
    'phone': {...} //first occurence
    'phone2': {...} //second ocurrence, automatically numbered
}
```

It's possible to add options to how this happens, like: dash separated, underline separated, no separation, etc.

Another would be to adopt an entire different scheme, appending other attributes such as [`autocomplete`](https://html.spec.whatwg.org/multipage/forms.html#autofill) [`itemprop`](https://en.wikipedia.org/wiki/Microdata_%28HTML%29) (microdata) or [`class`](https://en.wikipedia.org/wiki/Microformat) (microformat).

This way if the markup is something like this:

``` html
<form itemscope itemtype="http://schema.org/Person" class="vcard">
    <input type="tel" name="phone" required autocomplete="tel-work"> <!-- 1 -->
    <input type="tel" name="phone" itemprop="telephone"> <!-- 2 -->
    <input type="tel" name="phone" class="tel-type-cell"> <!-- 3 -->
</form>
```

The output would be like:

``` javascript
{
    'phone-tel-work': {...} //1
    'phone-telephone': {...} //2
    'phone-tel-type-cell': {...} //3
}
```

_Note_: the attributes example is way too convoluted, usually people choose one way of representing the data semantically and stick with it.

**Comment by dmitriytat** [comment]
@madcampos thank you for good ideas and the detailed answers!

**Comment by dmitriytat** [comment]
@madcampos i have a new idea. What do you think about it?

We can use array for defining riles like this

```
{
  'phone': [
    {required: true},
    {required: false},
    {}
  ],
  'email' : {required: true}
}
```

But i see one lack: need to specify rules for all (or almost all) inputs.

**Comment by madcampos** [comment]
That is good, better than having to add things to the names.
As for the repetition of the rules something can be worked out...
They could be extended, like extending the rules of `eslint`.

So with a code like this:

``` javascript
{
    'phone': [
        {
            'required': true,
            'regexp': /.*/
        },
        {
            'required': false,
            'extends': phone[0]
        }
    ]
}
```

We get the `regexp` field of the first object also being in the second object while preserving the `required` differences between the two objects.

The merging of the objects can be done with ES6's [`Object.assign()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)(with a example pollyfill) or lodash's [`_.assign`](https://lodash.com/docs/4.16.4#assign) or even jquery's [`$.extend()`](http://api.jquery.com/jQuery.extend/).
